packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -29,6 +29,18 @@ enum FlexDirection {
   vertical
 }
 
+/// Which incoming constraint parameter defines how much free space is
+/// available along the main axis in a flex layout.
+enum MainAxisSpace {
+  /// The free space is the difference between incoming constraint's min parameter
+  /// and the collective size of the children, but not less than zero.
+  min,
+
+  /// The free space is the difference between incoming constraint's max parameter
+  /// and the collective size of the children, but not less than zero.
```

Comment:
It can totally be less than zero.

If we frame this as "MainAxisSize" rather than "MainAxisSpace", we can explain this as "the size in the main axis is the greater of the incoming constraint's min parameter, and the collective size of the children" and "the size in the main axis is the incoming constraint's max parameter, unless that is infinite, in which case it is the collective size of the children" respectively.